### ISSUE TITLE
Better error handling for wrong URL for client

### ISSUE THREAD

**Issue by mvadari** [XRPLF/xrpl-py#353]
If you put in the wrong URL of a node for a client, the error that it returns is super opaque (it's a JSONDecodeError). We should return a better error/error message, so that it's clear that the user entered the wrong URL for the node.

**Comment by khancode** [comment]
@mvadari Can you share code to repro this?

**Comment by mvadari** [comment]
Here's one example:
```
>>> client = JsonRpcClient("https://s2.ripple.com:51233")
>>> client.request(ServerInfo())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mvadari/Documents/xrpl-py/xrpl/clients/sync_client.py", line 28, in request
    return asyncio.run(self.request_impl(request))
  File "/Users/mvadari/.pyenv/versions/3.9.1/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Users/mvadari/.pyenv/versions/3.9.1/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/Users/mvadari/Documents/xrpl-py/xrpl/asyncio/clients/json_rpc_base.py", line 39, in request_impl
    return json_to_response(response.json())
  File "/Users/mvadari/Documents/xrpl-py/.venv/lib/python3.9/site-packages/httpx/_models.py", line 1415, in json
    return jsonlib.loads(self.text, **kwargs)
  File "/Users/mvadari/.pyenv/versions/3.9.1/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Users/mvadari/.pyenv/versions/3.9.1/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Users/mvadari/.pyenv/versions/3.9.1/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

**Comment by mvadari** [comment]
Resolved by #388
